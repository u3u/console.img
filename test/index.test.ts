// @vitest-environment happy-dom
import { expect, it, vi } from 'vitest'
import consoleImg from '../src'

it('The `console.img` should be defined', () => {
  expect(console.img).toBeDefined()
  expect(console.img).toBe(consoleImg)
})

it('Should match snapshots', async () => {
  const url =
    'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+M9QDwADhgGAWjR9awAAAABJRU5ErkJggg=='

  vi.mock('../src/load-image.ts', () => {
    return {
      loadImage: vi.fn(() => {
        const img = new Image(400, 300)

        // @ts-expect-error
        img.naturalWidth = img.width
        // @ts-expect-error
        img.naturalHeight = img.height

        return img
      }),
    }
  })

  const spy = vi.spyOn(console, 'log')

  await consoleImg(url, { fit: 'cover', width: 100 })
  await consoleImg(url, { height: 100, styles: ['background-size: 100%'] })
  await consoleImg(url, { debug: true, styles: 'background-size: 100% 100%' })

  expect(spy.mock.calls).toMatchSnapshot()
})
