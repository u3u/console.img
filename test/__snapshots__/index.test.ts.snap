// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should match snapshots 1`] = `
[
  [
    "%c ",
    "font-size: 0; padding: 37.5px 50px; background: url(data:image/png;base64,) no-repeat 50% / cover",
  ],
  [
    "%c ",
    "font-size: 0; padding: 50px 66.66666666666666px; background: url(data:image/png;base64,) no-repeat 50% / contain; background-size: 100%",
  ],
  [
    "%c ",
    "font-size: 0; padding: 150px 200px; background: url(data:image/png;base64,) no-repeat 50% / contain; background-size: 100% 100%",
  ],
  [
    "[console.img]",
    "3.0.6",
    [
      "font-size: 0",
      "padding: 150px 200px",
      "background: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNk+M9QDwADhgGAWjR9awAAAABJRU5ErkJggg==) no-repeat 50% / contain",
      "background-size: 100% 100%",
    ],
  ],
]
`;
